src/BinaryParsers/MachOBinary/MachOBinary.cs

```diff
@@ -59,5 +75,148 @@ public static bool CanLoadBinary(Uri uri)
+                if (section != null)
+                {
+                    codeSegmentOffset = seg.Address - seg.FileOffset;
+                    break;
+                }
+            }
+
+            return section != null ? address - codeSegmentOffset : 0;
+        }
+
+        public string GetDwarfCompilerCommand()
+        {
+            if (CompilationUnits == null || CompilationUnits.Count == 0)
+            {
+                return string.Empty;
+            }
+            KeyValuePair<DwarfAttribute, DwarfAttributeValue> producer = CompilationUnits
+                .SelectMany(c => c.Symbols)
+                .Where(s => s.Tag == DwarfTag.CompileUnit)
+                .SelectMany(s => s.Attributes)
+                .FirstOrDefault(a => a.Key == DwarfAttribute.Producer);
+            return producer.Key == DwarfAttribute.None ? string.Empty : producer.Value.String;
+        }
+
+        byte[] IDwarfBinary.DebugData => throw new NotImplementedException();
+
+        byte[] IDwarfBinary.DebugDataDescription => throw new NotImplementedException();
+
+        byte[] IDwarfBinary.DebugDataStrings => throw new NotImplementedException();
+
+        byte[] IDwarfBinary.DebugLine => throw new NotImplementedException();
+
+        byte[] IDwarfBinary.DebugFrame => throw new NotImplementedException();
+
+        byte[] IDwarfBinary.EhFrame => throw new NotImplementedException();
+
+        ulong IDwarfBinary.CodeSegmentOffset => throw new NotImplementedException();
+
+        ulong IDwarfBinary.EhFrameAddress => this.GetSectionAddress(SECTIONNAME_EH_FRAME);
+
+        ulong IDwarfBinary.TextSectionAddress => this.GetSectionAddress(SECTIONNAME_TEXT);
+
+        ulong IDwarfBinary.DataSectionAddress => this.GetSectionAddress(SECTIONNAME_DATA);
+
+        IReadOnlyList<DwarfPublicSymbol> IDwarfBinary.PublicSymbols => throw new NotImplementedException();
+
+        bool IDwarfBinary.Is64bit => this.MachO.Is64;
+
+        #endregion
+
+        private byte[] LoadSection(string sectionName)
+        {
+            Section section = this.Segments.SelectMany(seg => seg.Sections.ToList())
+                                           .Where(sec => sec.Name == sectionName || sec.Name == sectionName + ".dwo")
+                                           .FirstOrDefault();
+
+            return section != null ? section.GetData() : Array.Empty<byte>();
+        }
+
+        private ulong GetSectionAddress(string sectionName)
+        {
+            foreach (Segment segment in this.Segments)
+            {
+                Section section = segment.Sections.Where(sec => sec.Name == sectionName).FirstOrDefault();
+                if (section != null)
+                {
+                    ulong CodeSegmentOffset = segment.Address - (ulong)segment.FileOffset;
+                    ulong LoadOffset = 0;
+                    ulong loadOffset = ((0 /*section.Flags*/ & 0x2) > 0) ? LoadOffset : 0;

```

Comment:
is this correct?

Reply:
the ELF version checks if section flag is Allocatable 0x2, which not applicable for Mach-O section, its always 0. I will refacor. Its used to the address value of sections __text/__data/__ef_frame . These addressese are set to property value and not used to access the content. In the tests these address looks good but don't have a way to verify so far.